modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/snapshot/outgoing/OutgoingSnapshotMetricsSource.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.partition.replicator.raft.snapshot.outgoing;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.metrics.AbstractMetricSource;
+import org.apache.ignite.internal.metrics.AtomicIntMetric;
+import org.apache.ignite.internal.metrics.AtomicLongMetric;
+import org.apache.ignite.internal.metrics.Duration;
+import org.apache.ignite.internal.metrics.Metric;
+import org.apache.ignite.internal.metrics.StringGauge;
+import org.apache.ignite.internal.partition.replicator.raft.snapshot.PartitionKey;
+import org.apache.ignite.internal.partition.replicator.raft.snapshot.incoming.IncomingSnapshotCopier;
+import org.apache.ignite.internal.partition.replicator.raft.snapshot.outgoing.OutgoingSnapshotMetricsSource.Holder;
+import org.apache.ignite.internal.raft.RaftGroupConfiguration;
+
+/**
+ * Tracks important metrics for the outgoing snapshot.
+ *
+ * <p>Note: not thread safe. Thread safe guaranteed by operation's order in {@link IncomingSnapshotCopier}.
+ */
+public class OutgoingSnapshotMetricsSource extends AbstractMetricSource<Holder> {
+    private static final IgniteLogger LOG = Loggers.forClass(OutgoingSnapshotMetricsSource.class);
+
+    private final Duration currentBatchDuration = new Duration();
+
+    private long combinedBatchDuration;
+
+    private final Duration totalSnapshotDuration = new Duration();
+
+    private final List<String> peers = new ArrayList<>();
+
+    private final List<String> learners = new ArrayList<>();
+
+    private final UUID snapshotId;
+
+    private final PartitionKey partitionKey;
+
+    /** Constructor. */
+    public OutgoingSnapshotMetricsSource(UUID snapshotId, PartitionKey partitionKey) {
+        super("snapshots.outgoing." + snapshotId);
```

Comment:
Let's specify that it's about Raft snapshots, to avoid confusion. `raft.snapshots.outgoing`?